runtime/lua/vim/lsp.lua

```diff
@@ -26,6 +27,11 @@ local lsp = {
 -- TODO consider whether 'eol' or 'fixeol' should change the nvim_buf_get_lines that send.
 -- TODO improve handling of scratch buffers with LSP attached.
 
+local function err_message(...)
+  nvim_err_writeln(table.concat(vim.tbl_flatten{...}))
```

Comment:
TODO (for myself): future `nvim_message` function should have "truncate" flag. I e show only the first 'cmdheight' line[s] in message area + some indicator, but store the full message in history.